### DISCUSSION TITLE
check_date:smoothness

### DISCUSSION THREAD

**Discussion by fabiobettani** [streetcomplete/StreetComplete#5362] [Ideas]
Lately I have been tagging the smoothness of road surfaces in my area, mostly with cycling routing in mind. But I'm afraid that this information could be subject to quality decay over time, as roads get renewed periodically and mappers typically don't.

It feels like this is the kind of tag where it would be useful if StreetComplete could automatically add a "check_date:smoothness" parameter, so that the question can be renewed once a given amount of time (e.g. 2 years?) has passed. Is this doable?

**Comment by jonpsp** [comment]
I agree - roads in my area deteriorate due to lack of maintenance, so the value should be reviewed regularly.

**Comment by mnalis** [comment]
It is already being re-checked every 4 years, e.g.:

https://github.com/streetcomplete/StreetComplete/blob/512afdc8c7bc6c58086a12471b5b1df4e47649c5/app/src/main/java/de/westnordost/streetcomplete/quests/smoothness/AddRoadSmoothness.kt#L20-L21

You can halve (or double) that time in `Settings` / `Resurvey intervals`

**Reply by fabiobettani** [reply]
Thanks, this looks very interesting. Sorry for asking, but I'm not sure about this: does this mean that StreetComplete is able to evaluate when the parameter was last edited on OSM? Or is it simply 4 years since *I* last edited the parameter through StreetComplete?

**Reply by FloEdelmann** [reply]
* If element has no `smoothness` tag yet
  → quest will be shown
* If element's last modification date is more than 4 years ago
  → quest will be shown
* If element's last modification date is less than 4 years ago and element has no `smoothness:check_date` tag
  → quest will not be shown
* If element's last modification date is less than 4 years ago and element has `smoothness:check_date` tag (with value less than 4 years ago)
  → quest will not be shown
* Element's last modification date is less than 4 years ago and element has `smoothness:check_date` tag (with value more than 4 years ago)
  → quest will be shown

When answering the quest:

* If smoothness answer is the same as already tagged
  → add `smoothness:check_date` tag with today's date
* If smoothness answer changes
  → change `smoothness` tag value accordingly and remove `smoothness:check_date` tag

**Reply by mnalis** [reply]
@fabiobettani as explained by @FloEdelmann in detail above, the short answer in practice is _"yes, StreetComplete is able to evaluate when the `smoothness` was last edited on OSM (by anyone), and ask for resurvey automatically if it hasn't been resurveyed in a long time"_
